skimage/graph/_graph.py

```diff
@@ -72,8 +75,9 @@ def pixel_graph(image, *, mask=None, edge_function=None, connectivity=1, spacing
     """
     if image.dtype == bool and mask is None:
         mask = image
-    if mask is None and edge_function is None:
+    if mask is None:
         mask = np.ones_like(image, dtype=bool)
+    if edge_function is None:
         edge_function = _weighted_abs_diff
```

Comment:
Tests pass with this change, which keeps the original behaviour intended with `edge_function`:
```suggestion
if mask is None:
    if image.dtype == bool:
        mask = image
    else:
        mask = np.ones_like(image, dtype=bool)
        edge_function = _weighted_abs_diff
```

Reply:
And revert b2d9fed6cf8343b284a1b0c9c484d9f464c7b9e4.

Reply:
Thank you for the suggestion. Let me change it as suggested.

Reply:
You're very welcome! Fyi, you may incorporate suggestions via GitHub's interface (also in *batches* when there are multiple suggestions): https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request